### DISCUSSION TITLE
Why is getServerSideProps() executed on every page navigation?

### DISCUSSION THREAD

**Discussion by JorgeSivil** [vercel/next.js#11870] [Help]
It was my understanding that getServerSideProps() is meant for the server side rendering phase, then React would do as normal as in a SPA.

I am a bit confused.

How do I achieve what I want? I don't like having a lot of ajax calls exposing my serverSideProps as they are supposed to be server side only.

Can you clear my mind a bit?

**Comment by timneutkens** [comment]
The properties are always sent to the browser, even when server-rendering as they are needed for hydration.

The behavior of `getServerSideProps` is explained here: https://nextjs.org/docs/basic-features/data-fetching#only-runs-on-server-side

It runs server-side but you still need access to what's returned client-side.

`getServerSideProps` is not specifically tied to server-rendering, it only implies it runs server-side. On initial request that means before rendering. With client-side navigation through `next/link` that means it fetches an API endpoint that executes `getServerSideProps` automatically

**Reply by LNFWebsite** [reply]
I have a question along these lines. How do you import a function that is supposed to be purely server-side for use in `getServerSideProps`? Because I can't import a function which imports `fs` in the top of the page. It treats it as client-side code.

**Reply by JorgeSivil** [reply]
You have to do a dynamic import

e.g. in my code base

```
const ContextSeeder = (await import('lib/Context/ContextSeeder')).default;
const props = await new ContextSeeder(context).withCollectionPage().withUser().get();
```

**Reply by LNFWebsite** [reply]
Thank you that did the trick!

**Reply by devknoll** [reply]
@LNFWebsite that sounds like a bug. Can you open an issue with a reproduction?

**Comment by jasonwilliams** [comment]
@timneutkens what do you recommend for needing to call certain functions before firing `getServerSideProps` ? Right now we wrap every page in a HOC, then that does the relevant checks before we use the Page's `getInitialProps` (_is the user logged in?_ for example). This model doesn't really work when switching to functional components and `getServerSideProps`. Do you have any advice?
